Library/Homebrew/load_path.rb

```diff
@@ -5,12 +5,20 @@
 
 HOMEBREW_LIBRARY_PATH = Pathname(__dir__).realpath.freeze
 
-$LOAD_PATH.push HOMEBREW_LIBRARY_PATH.to_s
+require_relative "utils/gems"
+Homebrew.setup_gem_environment!(setup_path: false)
 
-require "vendor/bundle/bundler/setup"
-require "homebrew_bootsnap"
+$LOAD_PATH.push HOMEBREW_LIBRARY_PATH.to_s
+require_relative "vendor/bundle/bundler/setup"
+$LOAD_PATH.uniq!
```

Comment:
I moved this to before Bootsnap loading and made it unconditional as a result. Let me know if it still breaks Bootsnap - it seems to work on Linux portable ruby.

Reply:
It's more than Bootsnap renders it unnecessary. Why did it need moved before, again? Not opposed, just want to understand.

Reply:
Oh, right. The line was originally added to avoid Ruby crashes (like the above) caused by a too large `$LOAD_PATH`. The bundler setup file adds a load of items to the `$LOAD_PATH` so I try prune that before subsequent requires.

Basically, stuff like our integration tests would duplicate the `$LOAD_PATH` and cause it to massively increase. I'm working on another commit now however to try stop that behaviour at source.

Reply:
Fine with me 👍🏻